### PR TITLE
refactor: remove app manifest in Context

### PR DESCRIPTION
**Pull request by jayzhang**
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/10420874

### CODE REVIEW

File: packages/fx-core/src/plugins/resource/appstudio/plugin.ts [hunk 1]
```diff
@@ -502,7 +502,9 @@ export class AppStudioPluginImpl {
       }
     } else {
       appDirectory = `${ctx.root}/.${ConfigFolderName}`;
-      manifestString = JSON.stringify(ctx.app);
+      manifestString = (
+        await fs.readFile(`${ctx.root}/.${ConfigFolderName}/${REMOTE_MANIFEST}`)
```
Comment (nliu-ms) [inline packages/fx-core/src/plugins/resource/appstudio/plugin.ts:506]: for azure project, we need to generate remote manifest, not just read manifest file.

File: packages/fx-core/tests/plugins/resource/function/unit/scaffold.test.ts [hunk 2]
```diff
@@ -43,6 +43,16 @@ const context: any = {
   root: path.join(__dirname, "ut"),
 };
 
+context.projectSettings = {
```
Comment (hund030) [inline packages/fx-core/tests/plugins/resource/function/unit/scaffold.test.ts:46]: Seems this setting isn't used by function ut. Could you help confirm and remove this if we don't need?

**Review by KennethBWSong — APPROVED** [review]

**Review by YitongFeng-git — APPROVED** [review]

**Review by 1yefuwang1 — APPROVED** [review]

**Review by hund030 — APPROVED** [review]

File: packages/fx-core/src/plugins/resource/appstudio/plugin.ts [hunk 3]
```diff
@@ -328,27 +329,27 @@ export class AppStudioPluginImpl {
...
+    // if (maybeSelectedPlugins.isErr()) {
+    //   return err(maybeSelectedPlugins.error);
+    // }
+    // const selectedPlugins = maybeSelectedPlugins.value;
+    // if (selectedPlugins.some((plugin) => plugin.name === "fx-resource-bot")) {
+    const capabilities = (ctx.projectSettings?.solutionSettings as AzureSolutionSettings)
+      .capabilities;
+    const hasBot = capabilities?.includes(BotOptionItem.id);
+    const hasMsgExt = capabilities?.includes(MessageExtensionItem.id);
+    if (!hasBot && !hasMsgExt) {
```
Comment (nliu-ms) [inline packages/fx-core/src/plugins/resource/appstudio/plugin.ts:344]: The original logic throw error when bot plugin is selected without bot/me capability.

But here, it throws error when the app is not bot/me app.

File: packages/fx-core/src/plugins/resource/appstudio/plugin.ts [hunk 4]
```diff
@@ -1137,28 +1151,8 @@ export class AppStudioPluginImpl {
...
-            AppStudioError.PluginNotFound.name,
-            AppStudioError.PluginNotFound.message(name)
-          )
-        );
-      }
-      results.push(plugin);
-    }
-    return ok(results);
+    const plugins = getActivatedResourcePlugins(azureSettings);
+    azureSettings.activeResourcePlugins = plugins.map((p) => p.name);
```
Comment (nliu-ms) [inline packages/fx-core/src/plugins/resource/appstudio/plugin.ts:1155]: seems the assignment to azureSettings is not used anymore.

getSelectedPlugins equals getActivatedResourcePlugins here.

Comment (jayzhang) [reply packages/fx-core/src/plugins/resource/appstudio/plugin.ts:1155]: Yes, just copy from solution. This method can be deleted

**Review by nliu-ms — APPROVED** [review]

**Pull request merged**